src/Microsoft.DotNet.Watcher.Core/Internal/FileWatcher/FileWatcherFactory.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.DotNet.Watcher.Core.Internal
+{
+    public static class FileWatcherFactory
+    {
+        public static IFileSystemWatcher CreateWatcher(string watchedDirectory)
+        {
+            var envVar = Environment.GetEnvironmentVariable("USE_POLLING_FILE_WATCHER");
```

Comment:
It has nothing to do with network share. It's just enables a polling file watcher which can be used for non-network shares. Also, it's not an ASPNETCORE feature (at least in dotnet-watch)